src/glview/VBORenderer.cc

```diff
@@ -749,16 +751,36 @@ void VBORenderer::add_shader_pointers(VertexArray& vertex_array)
     });
   }
 
+  if (getShader().data.csg_rendering.marked) {
```

Comment:
In general, it feels like the `marked` concept is complicating a bunch of code. Would it be possible to distinguish between marked and non-marked polygons by collecting them in different VBOs instead, and rendering them using different shaders, instead of forcing more complex shaders managing an extra vertex attribute?